lib/Dialect/HW/HWOps.cpp

```diff
@@ -1710,6 +1710,16 @@ LogicalResult ArrayCreateOp::verify() {
   return success();
 }
 
+Value ArrayCreateOp::getUniformElement() {
+  if (getInputs().empty())
+    return {};
+  auto first = getInputs()[0];
+  if (llvm::all_of(getInputs().drop_front(),
```

Comment:
Consider `llvm::all_equal(getInputs())`?

Reply:
:raised_hands: Awesome, I did not know that that thing existed. Very useful!

Reply:
Done